### ISSUE TITLE
TypeError: func is not a function

### ISSUE THREAD

**Issue by kinyat** [elastic-coders/serverless-webpack#41]
Sorry I get stuck for a while already so i would really love to have a hand on this issue.
serverless-webpack works when i run `sls deploy` to the cloud but not for the local server.
I guess i must have missed something.
```
$ sls webpack serve
Serverless: Serving functions...
  GET - http://localhost:8000/test
TypeError: func is not a function
  at /home/eric/development/first-serverless/node_modules/serverless-webpack/lib/serve.js:114:7
  at Layer.handle [as handle_request] (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/layer.js:95:5)
  at next (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/route.js:131:13)
  at Route.dispatch (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/route.js:112:3)
  at Layer.handle [as handle_request] (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/layer.js:95:5)
  at /home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:277:22
  at Function.process_params (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:330:12)
  at next (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:271:10)
  at jsonParser (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/body-parser/lib/types/json.js:103:7)
  at Layer.handle [as handle_request] (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:312:13)
  at /home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:330:12)
  at next (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:271:10)
  at expressInit (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/middleware/init.js:33:5)
  at Layer.handle [as handle_request] (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:312:13)
  at /home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:330:12)
  at next (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:271:10)
  at query (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/middleware/query.js:44:5)
  at Layer.handle [as handle_request] (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/layer.js:95:5)
  at trim_prefix (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:312:13)
  at /home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:280:7
  at Function.process_params (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:330:12)
  at next (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:271:10)
  at Function.handle (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/router/index.js:176:3)
  at EventEmitter.handle (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/application.js:173:10)
  at Server.app (/home/eric/development/first-serverless/node_modules/serverless-webpack/node_modules/express/lib/express.js:38:9)
  at emitTwo (events.js:87:13)
  at Server.emit (events.js:172:7)
  at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:528:12)
  at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
```

my package.json file
```
{
  "name": "first-serverless",
  "version": "0.0.1",
  "description": "First Serverless",
  "main": "handler.js",
  "dependencies": {
    "babel-runtime": "6.11.6",
    "babel-polyfill": "^6.16.0"
  },
  "devDependencies": {
    "aws-sdk": "^2.6.6",
    "babel-core": "^6.17.0",
    "babel-loader": "^6.2.5",
    "babel-plugin-transform-runtime": "^6.15.0",
    "babel-preset-es2015": "^6.16.0",
    "bluebird": "^3.4.6",
    "serverless-webpack": "^1.0.0-rc.2",
    "webpack": "^1.13.2",
    "webpack-node-externals": "^1.4.3"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Eric Chan",
  "license": "ISC"
}
```

webpack.config.js
```
var nodeExternals = require('webpack-node-externals');

module.exports = {
    entry: [
        'babel-polyfill',
        './handler.js'
    ],
    module: {
        loaders: [{
            test: /\.js$/,
            loaders: ['babel'],
            include: __dirname,
            exclude: /node_modules/,
        }]
    },
    target: 'node',
    externals: [nodeExternals()]
};
```

serverless.yml
```
service: aws-nodejs

provider:
  name: aws
  runtime: nodejs4.3

functions:
  hello:
    handler: handler.hello
    events:
      - http:
          path: test
          method: get

plugins:
  - serverless-webpack
custom:
  webpackIncludeModules: true
```

handler.js
```
export function hello (event, context, cb) {
    cb(null, { message: 'Go Serverless v1.0! Your function executed successfully!', event });
}
```

**Comment by kinyat** [comment]
works after removing node_module and npm install...
pls close this ticket...

**Comment by kinyat** [comment]
Interestingly... 2nd call works...
But if i make any changes in the code and save it, the server will be terminated with the following Error.
It means I need to rerun sls webpack serve everytime when i have changes and I need to call the endpoint twice to see the expected result.
```
/home/eric/development/webpack-serverless/node_modules/babel-polyfill/lib/index.js:10
  throw new Error("only one instance of babel-polyfill is allowed");
  ^

Error: only one instance of babel-polyfill is allowed
    at Object.<anonymous> (/home/eric/development/webpack-serverless/node_modules/babel-polyfill/lib/index.js:10:9)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.defineProperty.value (/home/eric/development/webpack-serverless/.webpack/handler.js:55:19)
    at __webpack_require__ (/home/eric/development/webpack-serverless/.webpack/handler.js:20:30)
    at Object.<anonymous> (/home/eric/development/webpack-serverless/.webpack/handler.js:47:2)
    at __webpack_require__ (/home/eric/development/webpack-serverless/.webpack/handler.js:20:30)
    at /home/eric/development/webpack-serverless/.webpack/handler.js:40:18
    at Object.<anonymous> (/home/eric/development/webpack-serverless/.webpack/handler.js:43:10)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at ServerlessWebpack.loadHandler (/home/eric/development/webpack-serverless/node_modules/serverless-webpack/lib/run.js:19:20)
    at Watching.handler (/home/eric/development/webpack-serverless/node_modules/serverless-webpack/lib/serve.js:24:39)
    at Watching._done (/home/eric/development/webpack-serverless/node_modules/webpack/lib/Compiler.js:81:7)
```

**Comment by andymac4182** [comment]
I get that error if you call the endpoints before webpack has finished. I normally wait 30seconds after starting to ensure webpack has finished.

**Comment by utkuturunc** [comment]
I followed the babel example and I am also getting this error. 2nd call doesn't work for me

**Comment by thenikso** [comment]
Fixed this here e07ab1dcfac592c28f9fa9c7c6f4a64b0dff0fa3 basically the function reload code is not great (because it doesn't provide a fresh node environment when reloading functions). The fix proposed is more of a workaround of conditionally loading `babel-polyfill` in the handler rather than the webpack config. 

There might be better way to reload dependencies but this should hold for now.

**Comment by utkuturunc** [comment]
this solution requires the addition of babel-polyfill import to all files if multiple entry points are used.
